src/celengine/observer.cpp

```diff
@@ -1375,7 +1375,13 @@ Vector3f Observer::getPickRayFisheye(float x, float y) const
 // universal coordinates.
 void Observer::updateUniversal()
 {
+    UniversalCoord oldPositionUniv = positionUniv;
     positionUniv = frame->convertToUniversal(position, simTime);
+    if (positionUniv.clampRange())
+    {
+        positionUniv = oldPositionUniv;
```

Comment:
so the new position is not used at all? in this case i propose instead of clamping provide a method `isOutOfBounds` which will only check if the value is out of bounds.
on the other hand, what if observer position is updated by a script? in this case it will not move at all. maybe it will be better to "stop"  it near the "invisible wall"?

Reply:
Yeah, this is a bit of a remnant of my original approach which led to the observer sliding into the corner of the bounding cube, which felt really weird.